### ISSUE TITLE
Delete tags in Property Inspector

### ISSUE THREAD

**Issue by psychemedia** [jupyterlab/jupyterlab#11931]
In the cell Property Inspector, cells tags *for a particular notebook* can be added, enabled and disabled but not edited or deleted.

![image](https://user-images.githubusercontent.com/82988/150985908-51e780b5-bbc2-4d78-bb9d-e26e2176d3aa.png)

It would be be useful to be able to:

- select a tag and edit it;
- select a tag and delete it.

**Comment by krassowski** [comment]
Thank you for opening this issue. 

<details>
<summary>Clarification of the current state</summary>

If a tag exists on just one cell, clicking on the tag removes it:
 ![ezgif-7-24a5456e62](https://user-images.githubusercontent.com/5832902/151212069-169d2a2b-c839-41c1-aac7-e20577d62b2d.gif)

If tag exists on cell other than the selected cell, clicking on it enables/disables it but does not remove it altogether (which makes sense):
 ![Peek 2022-01-26 17-12](https://user-images.githubusercontent.com/5832902/151212507-56cce45b-0acd-4ebe-8726-534c9e2d1cb0.gif)

</details>

The current limitation is that we do not have a way to:
- remove a tag from all cells tagged with this tag
- fix a typo in an existing tag which exists on several cells

We need to define how to change the UI to allow for these two actions. Ideally we should also think how to make it work with #10354. Some quick ideas:

1. We could expose these actions in the **context menu**:
   - pro: keeps the UI simple
   - con: makes it difficult to discover
2. We could split the tag widget button into two/three clickable areas like: [✓ | tag name | ✗]
   - clicking on ✓ portion would enable/disable the tag
   - clicking on tag name would show a dialog to rename and warn that it will rename all instances of this tag
   - clicking on ✗ would show a dialog to ask if user wants to remove all instances of this tag

What do you think? Any other suggestions @psychemedia?

**Comment by psychemedia** [comment]
In the classic notebook, each tag has a delete `x` element, which has clear semantics:

![image](https://user-images.githubusercontent.com/82988/151325988-f6c5c1a0-9740-4544-a36f-3d61a31c45a0.png)

For editing, it would be simplest to just double click into the text of the tag and make it editable, then hit return to accept the change.

The ability to edit is not so important if you can easily delete a tag. (The classic notebook tag editor does not let you edit a tag one you have added it to a cell.)

**Comment by psychemedia** [comment]
I also hadn't appreciated the difference between how tags are handled when they are applied to once versus many cells. That's really confusing to me... I can see how it might be useful to know which tags are "common" in the sense of being used elsewhere in a particular notebook, as well as which tags are applied to a particular cell (which is likely the primary concern).

I could imagine an area where "available" tags at the notebook level are displayed, and tags used for a particular cell are displayed, but the mechanics of moving between them could be fiddly. Eg would I click an "available" tag to add it to a cell? Would I add a new tag in the cell area and then have it appear in the "available" area; or add new tags in the "available" area then click to add them to (one or more) selected cells?

In classic notebook, the tag view toggles to decorate each separate cell in the notebook view, but there is no "all used tags in the this notebook" display; whereas the JupyterLab view,  you are forced to view cell tags at just a single cell level, rendered in the context of a view of all the tags used in the notebook.

**Comment by krassowski** [comment]
Yes, the "available" area would prevent this kind of confusion. Alternatively, we could just have a completer-kind of suggestions showing the available tags when user clicks on the field for adding the tag (making it two clicks instead of a one click, but also providing an ability to filter the suggestions).

**Comment by echarles** [comment]
> I also hadn't appreciated the difference between how tags are handled when they are applied to once versus many cells. That's really confusing to me...

I also faced that confusion some time ago.

The main difference with the classic notebook is that jupyterlab has no ability to display multiple toolbars.

The toolbar capabilities are now discussed on https://github.com/jupyterlab/jupyterlab/issues/9733
